index.js

```diff
@@ -130,8 +144,8 @@ function IndyReq (conf) {
 
     let reqId = nextReqId++
     data.reqId = reqId
-
-    if (signKey) {
+
+    if (signKey && !("signatures" in data)) {
```

Comment:
Is it always going to be `signature` or `signatures`? Would there ever be a request with both a signature from the sender and other signatures by DID?

Reply:
If you have both `signature` and `signatures` the indy-node returns an error stating there should only be one. So I believe there will never be both of them in a single operation. I am in the process of filling a ticket around an issue with using only one signature in the `signatures` structure.

Reply:
It will always be `signature` or `signatures` but never both. When there are multiple signatures they all go into the `signatures` structure, including the author/sender signature. In many cases, the author/sender signature is required to be in the `signatures` structure. This is partially documented [here](https://github.com/hyperledger/indy-node/blob/master/docs/source/requests.md#common-write-request-structure).